### PR TITLE
[TestUtility] Add TestUtility plugin

### PR DESCRIPTION
**Pull request by artur-gebicz-ml**
TestUtility is a plugin, that allows to execute commands, to simulate
specific test conditions. This plugin can:
* Allocate a specific amount of memory
* Free previously allocated memory
* Retrieve information about currently allocated memory
* Crash plugin
* Crash plugin N times in a row

This plugin is meant to be easily extensible.

### CODE REVIEW

File: TestUtility/CommandCore/TestCommandBase.h [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+namespace WPEFramework {
+
+class TestCommandBase : public Exchange::ITestUtility::ICommand {
+    public:
+        class SignatureBuilder {
+            public:
+                SignatureBuilder(const SignatureBuilder&) = delete;
+                SignatureBuilder& operator=(const SignatureBuilder&) = delete;
+
```
Comment (pwielders) [inline TestUtility/CommandCore/TestCommandBase.h]: Be explicit, is the default constructor allowed ?

File: TestUtility/CommandCore/TestCommandBase.h [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+                }
+
+                TestCore::TestCommandSignature _jsonSignature;
+        };
+
+        class DescriptionBuilder {
+            public:
+                DescriptionBuilder(const DescriptionBuilder&) = delete;
+                DescriptionBuilder& operator=(const DescriptionBuilder&) = delete;
+
```
Comment (pwielders) [inline TestUtility/CommandCore/TestCommandBase.h]: Guess what, is the default constructor allowed ;-)

File: TestUtility/CommandCore/TestCommandBase.h [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                    _jsonDescription.ToString(outString);
+
+                    return outString;
+                }
+
+                TestCore::TestCommandDescription _jsonDescription;
+        };
+
+    private:
+        TestCommandBase(const TestCommandBase&) = delete;
```
Comment (pwielders) [inline TestUtility/CommandCore/TestCommandBase.h]: Although opinions have been different, I am in favor of being explicit and get difficult error messages, some people want to be counter intuitive :-) and put deleted constructors public. I am in favor of putting them private. I did not mention that before as I thought you where in the counter intuitive camp, but her you came to you senses :-D 
However You need to chose, move deleted constructors into a private section (as it should) or move them in the public section and delete. Here you opted for the right one, above you opted for the wron one. Please align.

File: TestUtility/CommandCore/TestCommandBase.h [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        TestCommandBase& operator=(const TestCommandBase&) = delete;
+
+    public:
+        explicit TestCommandBase(const DescriptionBuilder& description, const SignatureBuilder& signature)
```
Comment (pwielders) [inline TestUtility/CommandCore/TestCommandBase.h]: Uhhh how about the default constructor? Also default ? or deleted..

File: TestUtility/CommandCore/TestCommandController.cpp [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    static TestCommandController _singleton;
+    return (_singleton);
+}
+
+void TestCommandController::Announce(Exchange::ITestUtility::ICommand* command)
+{
+    ASSERT(command != nullptr);
+
+    _adminLock.Lock();
+    _commands[command->Name()] = command;
```
Comment (pwielders) [inline TestUtility/CommandCore/TestCommandController.cpp]: Maybe an assert in DEBUG, if an already existing command gets overrriden ? Or is that allowed and expected?

File: TestUtility/CommandCore/TestCommandBase.h (hunk 1 above)
Comment (artur-gebicz-ml) [reply TestUtility/CommandCore/TestCommandBase.h]: No, it shouldn't be allowed. Fixed.

File: TestUtility/CommandCore/TestCommandBase.h (hunk 2 above)
Comment (artur-gebicz-ml) [reply TestUtility/CommandCore/TestCommandBase.h]: No, it shouldn't be allowed. Fixed.

File: TestUtility/CommandCore/TestCommandBase.h (hunk 4 above)
Comment (artur-gebicz-ml) [reply TestUtility/CommandCore/TestCommandBase.h]: No, it shouldn't be allowed. Fixed.

File: TestUtility/CommandCore/TestCommandController.cpp (hunk 5 above)
Comment (artur-gebicz-ml) [reply TestUtility/CommandCore/TestCommandController.cpp]: Done

**Comment by woutermeek** [comment]
@artur-gebicz-ml could you resolve the CMakeLists.txt conflict so we can merge this?

**Pull request merged**